src/main/java/com/cloudbees/jenkins/support/impl/HeapHisto.java

```diff
@@ -0,0 +1,104 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.AsyncResultCache;
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.Content;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Node;
+import hudson.remoting.Callable;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import org.jenkinsci.remoting.RoleChecker;
+
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanException;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.ReflectionException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.management.ManagementFactory;
+import java.util.Collections;
+import java.util.Set;
+import java.util.WeakHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension
+public class HeapHisto extends Component {
+    private static final int OFFSET = 3;
+    private static final int MAX = 200 + OFFSET;
+
+    private static final Logger logger = Logger.getLogger(HeapHisto.class.getName());
+    private final WeakHashMap<Node, String> heapHistoCache = new WeakHashMap<Node, String>();
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Master Heap Histogram -histo:live";
```

Comment:
I a not sure what is this flag doing here. I would rather provide a more extensive documentation in help.jelly (if it is supported by the plugin).